lib/cloud_controller/opi/apps_client.rb

```diff
@@ -104,10 +104,35 @@ def to_hash
       end
     end
 
+    class KpackLifecycle
+      CNB_LAUNCHER_PATH = '/cnb/lifecycle/launcher'.freeze
+
+      def initialize(process)
+        @process = process
+      end
+
+      def to_hash
+        command = if @process.command.presence
+                    ['/bin/sh', '-c', CNB_LAUNCHER_PATH, @process.command]
+                  else
+                    []
+                  end
+        {
+          docker_lifecycle: {
```

Comment:
I suspect this was intentional and fine, but for my knowledge, why does this remain `docker_lifecycle` instead of something new like `kpack_lifecycle`? Is this because Eirini itself is looking for specific information under that key when it desires a LRP?

Reply:
Yeah, exactly.

Eirini is looking for either a `DockerLifecycle` or a `BuildpackLifecycle` (which I guess is only used in `kube-cf`):
https://github.com/cloudfoundry-incubator/eirini/blob/ee87b6612c6832f26a42a55ee309c222eee36bcb/bifrost/convert.go#L140

Reply:
Should we teach eirini to speak `KpackLifecycle`? Can merge this and revisit after if we really want to

Reply:
IMO we shouldn't until there is a reason to. I feel like at this point if Eirini doesn't need to distinguish between an app using a "regular" Docker image and an app using a kpack-built Docker image it shouldn't have to.